src/mtd.c

```diff
@@ -5,6 +5,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <mtd/mtd-user.h>
+#include <libgen.h>
```

Comment:
Rob, I think something’s missing from your patch.

Your commit message includes this

```
  128 |                        basename(strdup(mtd_path)));
```

showing that you’ve added a `strdup`, but that’s not present in this pull request ([here](https://github.com/robimarko/yafut/blob/b88a6ae187a33593ff74977ed1118db4305074aa/src/mtd.c#L128)).

With `#include <libgen.h>`, this is still incorrect without the `strdup`:

```
…/yafut/src/mtd.c: In function ‘read_oobavail_from_sysfs’:
…/yafut/src/mtd.c:135:33: warning: passing argument 1 of ‘__xpg_basename’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  135 |                        basename(mtd_path));
      |                                 ^~~~~~~~
In file included from …/yafut/src/mtd.c:7:
/usr/include/libgen.h:34:36: note: expected ‘char *’ but argument is of type ‘const char *’
   34 | extern char *__xpg_basename (char *__path) __THROW;
      |                              ~~~~~~^~~~~~
```

The `strdup` is necessary, and ideally the duplicated string will be `free`d after it’s used by `snprintf`.

Reply:
Also, I have moved to using the `miscellaneous-refactoring` branch which works without any patches for me

Reply:
Yes, thanks @robimarko for taking the time to prepare a pull request, but switching to the `miscellaneous-refactoring` branch is the way to go - I will merge its contents to `main` soon and I definitely don't want to spend any more time in the future on debugging someone else's build environment just to cater for a function that can be replaced with two lines of code.

Reply:
@kempniu Thanks, that sounds good.